sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ChangeFeedProcessorBuilder.java

```diff
@@ -150,6 +210,16 @@ private void validateChangeFeedProcessorBuilder() {
         if (leaseContainer == null) {
             throw new IllegalArgumentException("leaseContainer cannot be null");
         }
+        if ((partitionKeyBasedLeaseConsumer == null && epkRangeBasedLeaseConsumer == null)
+            || (partitionKeyBasedLeaseConsumer != null && epkRangeBasedLeaseConsumer != null)) {
+            throw new IllegalArgumentException("expecting either incremental or full fidelity consumer for handling change feed processor changes");
+        }
+        if (ChangeFeedMode.LATEST_VERSION.equals(changeFeedMode) && partitionKeyBasedLeaseConsumer == null) {
+            throw new IllegalArgumentException("consumer for handling change feed processor incremental changes cannot be null when using incremental mode");
```

Comment:
nit: change feed mode in the exception message

Reply:
Good catch, updated.